src/rt1_model/_calc.py

```diff
@@ -423,6 +423,8 @@ def set_geometry(self, t_0=None, p_0=None, t_ex=None, p_ex=None):
         set_bistatic : Use bistatic evaluation geometry.
 
         """
+        self._clear_geom_cache()
+
         if t_0 is not None:
```

Comment:
For better readability this could be a method like `_sanitize_geometry_parameter(parameter, error_message)` which handles the asserts and the numpy cast. Would also reduce code duplication.

Reply:
thanks, implemented!